File: greenfloor/daemon/testing/watchlist.py
"""Watchlist and active-offer counting patch points."""

from __future__ import annotations

from greenfloor.daemon.watchlist import (
    _active_offer_counts_by_size as active_offer_counts_by_size,
)
from greenfloor.daemon.watchlist import (
    _active_offer_counts_by_size_and_side as active_offer_counts_by_size_and_side,
)
from greenfloor.runtime.offer_watchlist import (
    build_dexie_size_by_offer_id,
    match_watched_coin_ids,
    set_watched_coin_ids_for_market,
    update_market_coin_watchlist_from_dexie,
)

__all__ = [
    "active_offer_counts_by_size",
    "active_offer_counts_by_size_and_side",
    "build_dexie_size_by_offer_id",
    "match_watched_coin_ids",
    "set_watched_coin_ids_for_market",
    "update_market_coin_watchlist_from_dexie",
]

File: greenfloor-engine/src/daemon/exports/watchlist.rs
pub use crate::daemon::coinset_tx::build_dexie_size_by_offer_id;
pub use crate::daemon::watchlist::{
    active_offer_counts_by_size, active_offer_counts_by_size_and_side,
    active_offer_counts_by_size_and_side_detail, active_offer_counts_by_size_detail,
    match_watched_coin_ids, set_watched_coin_ids_for_market, update_market_coin_watchlist_from_offers,
    watched_coin_ids_for_market, watchlist_offer_ids, CoinWatchlistCache,
    RESEED_MEMPOOL_MAX_AGE_SECONDS,
};
